crates/federated-server/src/server/engine_reloader.rs

```diff
@@ -139,24 +134,46 @@ async fn update_loop(
 }
 
 async fn build_new_engine(
-    current_config: gateway_config::Config,
+    config: gateway_config::Config,
     graph_definition: GraphDefinition,
     context: Context,
     operations_to_warm: Vec<OperationForWarming>,
 ) -> crate::Result<Arc<Engine<GatewayRuntime>>> {
-    let engine = gateway::generate(
-        graph_definition,
-        &current_config,
-        context.hot_reload_config_path,
-        context.hooks,
-    )
-    .await?;
+    let engine = gateway::generate(graph_definition, &config, context.hot_reload_config_path, context.hooks).await?;
 
     let engine = Arc::new(engine);
 
-    if current_config.operation_caching.warm_on_reload {
-        engine.warm_operation_cache(operations_to_warm).await;
-    }
+    engine.warm_operation_cache(operations_to_warm).await;
 
     Ok(engine)
 }
+
+fn extract_operations_to_warm(
+    config: &gateway_config::Config,
+    engine_sender: &EngineSender,
+) -> Vec<OperationForWarming> {
+    if !config.operation_caching.enabled || !config.operation_caching.warm_on_reload {
+        return vec![];
+    }
+
+    let (mut operations, cache_count) = {
+        let cache = &engine_sender.borrow().runtime.operation_cache;
+
+        (
+            cache.values().map(OperationForWarming::new).collect(),
+            cache.entry_count(),
+        )
+    };
+
+    if config.operation_caching.warming_percent >= 100 {
+        return operations;
+    }
+
+    // If we're not taking 100% of the list we shuffle the list and take a percentage of it
+    operations.shuffle(&mut thread_rng());
```

Comment:
No idea whether the current LRU cache makes it possible, but with `warming_percent` I would have expected the _top_ X% to be warmed, not a random subset of the cache. So would change the the documentation of the `gateway-config` to make it more obvious IMHO.

Reply:
Not even sure to see why you're shuffling at all to be honest.

Reply:
> Not even sure to see why you're shuffling at all to be honest if having a random subset is the goal.

What would the alternative be?

Reply:
Well ok, you do get a higher probably that a query is cached by at least one worker supposing that they receive in the same/similar order. But as load is balanced across your worker they will each still need to replan the missing 90% if you're not using shared cache.

I'm just kinda mitigated of the usefulness of replanning a random % of plans. You can't predict in any way how much of your traffic will stay cached. You might end up with 90% traffic cached or 0.1% if your traffic.